examples/features/scale-from-zero/README.md

```diff
@@ -0,0 +1,94 @@
+# Test automatic scale from zero
+
+This example shows that NSEs can be created on the fly, allowing effective scaling by the node.
+
+## Run
+
+Create test namespace:
+```bash
+NAMESPACE=($(kubectl create -f ../namespace.yaml)[0])
+NAMESPACE=${NAMESPACE:10}
+```
+
+Register namespace in `spire` server:
+```bash
+kubectl exec -n spire spire-server-0 -- \
+/opt/spire/bin/spire-server entry create \
+-spiffeID spiffe://example.org/ns/${NAMESPACE}/sa/default \
+-parentID spiffe://example.org/ns/spire/sa/spire-agent \
+-selector k8s:ns:${NAMESPACE} \
+-selector k8s:sa:default
+```
+
+Create customization file:
+```bash
+cat > kustomization.yaml <<EOF
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: ${NAMESPACE}
+
+bases:
+- ../../../apps/nse-supplier-k8s
+- ../../../apps/nsc-kernel
+
+patchesStrategicMerge:
+- patch-nsc.yaml
+- patch-supplier.yaml
+EOF
+```
+
+Register network service:
+```bash
+kubectl apply -f scale-ns.yaml
+```
+
+Deploy NSC and supplier:
+```bash
+kubectl apply -k .
+```
+
+Wait for applications ready:
+```bash
+kubectl wait --for=condition=ready --timeout=1m pod -l app=nse-icmp-responder -n ${NAMESPACE}
```

Comment:
Why do we need to wait?

Reply:
On a fresh cluster start of the pods can time quite long time. If we don't wait, subsequent commands will fail.

Reply:
we have a prefetch for this. This should not affect the test.

Reply:
All of the other tests have wait till pods are ready. Are you sure it's safe to remove the wait?
Also, if someone replicates this example manually, they will have to wait manually until all the images are downloaded.

This wait doesn't increase duration of the test. If we already have everything in the cache, and pods started immediately, `kubectl wait --for=condition=ready` will return without any delay.